### PR TITLE
The  BarChart

### PR DESCRIPTION
**Pull request by emer289**
## Purpose

The problem was to create a barchart to visualise data.
https://github.com/dabreadman/Node-Podman-OpenShift-CI-CD/issues/43#issue-833109404

## Proposed Change

I made a BarChart for the stats page

## Checklist

-[] Have code for the BarChart
-[] Display data on the BarChart

## Additional

I couldn't get it to work in the .js file.

## Learning

I had to familiarise myself with javascript and html to create the Bar chart.

https://www.hostinger.com/tutorials/what-is-html#:~:text=HTML%20stands%20for%20Hypertext%20Markup,for%20web%20pages%20and%20applications.&text=When%20working%20with%20HTML%2C%20we,mark%20up%20a%20website%20page.
https://www.tutorialspoint.com/javascript/javascript_overview.htm#:~:text=JavaScript%20is%20a%20dynamic%20computer,language%20with%20object%2Doriented%20capabilities.
https://websiddu.com/blog/how-to-make-a-bar-chart-using-vue-js.html#goals

### CODE REVIEW

File: covid-vue-app/src/pages/StatsPage.vue [hunk 1]
```diff
@@ -1,16 +1,121 @@
...
+  
+    computed: {
+      yTicks() {
+        return this.y.ticks(this.tickCount);
+      },
+      x() {
+        return scaleBand()
+          .range([0, this.width])
+          .padding(this.barPadding)
+          .domain(this.dataSet.map(e => e[0]));
```
Comment (saisankp) [inline covid-vue-app/src/pages/StatsPage.vue:88]: Maybe change line 88 to .domain(this.dataSet.map( (e) => e[0])); 
(Including brackets around the "e", to help with Codacy)

File: covid-vue-app/src/pages/StatsPage.vue [hunk 2]
```diff
@@ -1,16 +1,121 @@
...
+      },
+      y() {
+        let values = this.dataSet.map(e => e[1]);
```
Comment (saisankp) [inline covid-vue-app/src/pages/StatsPage.vue:91]: Maybe change line 91 to let values = this.dataSet.map( (e) => e[1]);
(Including brackets around the "e", to help with Codacy)

File: covid-vue-app/src/pages/StatsPage.vue [hunk 3]
```diff
@@ -1,16 +1,121 @@
...
+        return scaleLinear()
+          .range([this.height, 0])
+  
+          .domain([0, Math.max(...values)]);
+      },
+      bars() {
+        let bars = this.dataSet.map(d => {
```
Comment (saisankp) [inline covid-vue-app/src/pages/StatsPage.vue:98]: Maybe change line 91 to let bars = this.dataSet.map( (d) => {
(Including brackets around the "d", to help with Codacy)

**Review by saisankp — CHANGES_REQUESTED** [review]

File: covid-vue-app/package.json [hunk 4]
```diff
@@ -22,7 +25,7 @@
     "@vue/cli-plugin-eslint": "^4.5.12",
     "@vue/cli-service": "^4.5.12",
     "babel-eslint": "^10.0.1",
-    "eslint": "^7.23.0",
```
Comment (dabreadman) [inline covid-vue-app/package.json:25]: Can you bump `eslint` version?

File: covid-vue-app/src/components/Home.vue [hunk 5]
```diff
@@ -0,0 +1,23 @@
+<template>
+  <div class="home">
```
Comment (dabreadman) [inline covid-vue-app/src/components/Home.vue:2]: Why is this named `Home.vue`? It's it just chart?

File: covid-vue-app/src/pages/StatsPage.vue [hunk 6]
```diff
@@ -1,16 +1,121 @@
 <template lang="html">
   <v-container class="py-12 height_class">
     <h1>This is the Stats page</h1>
+    <svg class="barchart" :width="width + marginLeft / 2" :height="height + marginTop">
```
Comment (dabreadman) [inline covid-vue-app/src/pages/StatsPage.vue:4]: Make a new component called `BarChart` in [here](https://github.com/dabreadman/Node-Podman-OpenShift-CI-CD/tree/emer_visualisedata/covid-vue-app/src/components).

Check out the other components to see what you have to do.

**Comment by dabreadman** [comment]
What is this..
> ## Checklist
> -[] Have code for the BarChart
> -[] Display data on the BarChart

Do this instead
```
- [x] something finished
```

File: covid-vue-app/src/components/Home.vue (hunk 5 above)
Comment (emer289) [reply covid-vue-app/src/components/Home.vue:2]: I don't need it so I'll delete it

File: covid-vue-app/src/pages/StatsPage.vue (hunk 6 above)
Comment (emer289) [reply covid-vue-app/src/pages/StatsPage.vue:4]: I've added this and it works with yarn serve but doesn't work with docker-compose up

File: covid-vue-app/src/pages/StatsPage.vue (hunk 1 above)
Comment (emer289) [reply covid-vue-app/src/pages/StatsPage.vue:88]: I've done this thanks

File: covid-vue-app/src/pages/StatsPage.vue (hunk 6 above)
Comment (dabreadman) [reply covid-vue-app/src/pages/StatsPage.vue:4]: Do you have the logs?
If I remember correctly it's something about dependencies?

**Comment by dabreadman** [comment]
```
..`94`
-  Building for production...
 ERROR  Failed to compile with 1 error11:53:27 AM

 error  in ./node_modules/vuetify/src/components/VGrid/_grid.sass

Error: EISDIR: illegal operation on a directory, read
..`99`
```
Could you do `yarn build` locally?

**Comment by dabreadman** [comment]
Duplicate, abandoned.

**Pull request closed without merging**